src/main/groovy/com/netflix/gradle/plugins/application/OspackageApplicationSpringBootPlugin.groovy

```diff
@@ -54,85 +53,95 @@ class OspackageApplicationSpringBootPlugin implements Plugin<Project> {
-            project.afterEvaluate {
-                project.tasks.named('bootJar').configure {
-                    if (GradleVersion.current().baseVersion < GradleVersion.version('6.0').baseVersion) {
-                        classifier = 'boot'
-                    } else {
-                        archiveClassifier = 'boot'
-                    }
+        project.plugins.withId("org.springframework.boot") {
+            // Spring Boot 2.0+ configured distributions that have everything we need
+            if (project.distributions.findByName('boot') != null) {
+                project.logger.info("Spring Boot 2+ detected")
+                // Use the main distribution and configure it to have the same baseName as the boot distribution
+                project.tasks.named(JavaPlugin.JAR_TASK_NAME) {
+                    enabled = true
                 }
-                project.distributions {
-                    main {
+                project.afterEvaluate {
+                    project.tasks.named('bootJar').configure {
                         if (GradleVersion.current().baseVersion < GradleVersion.version('6.0').baseVersion) {
-                            baseName = "${project.distributions.main.baseName}-boot"
+                            classifier = 'boot'
                         } else {
-                            getDistributionBaseName().set "${project.distributions.main.getDistributionBaseName().getOrNull()}-boot"
+                            archiveClassifier = 'boot'
+                        }
+                    }
+                    project.distributions {
+                        main {
+                            if (GradleVersion.current().baseVersion < GradleVersion.version('6.0').baseVersion) {
+                                baseName = "${project.distributions.main.baseName}-boot"
+                            } else {
+                                getDistributionBaseName().set "${project.distributions.main.getDistributionBaseName().getOrNull()}-boot"
+                            }
                         }
                     }
-                }
 
-                // Allow the springBoot extension configuration to propagate to the application plugin
-                def mainClass = project.objects.property(String)
-                try {
-                    mainClass.set(project.springBoot.mainClass)
-                } catch (Exception ignore) {
-                    mainClass.set(project.springBoot.mainClassName)
-                }
-                if (!mainClass.isPresent()) {
+                    // Allow the springBoot extension configuration to propagate to the application plugin
+                    def mainClass = project.objects.property(String)
                     try {
-                        mainClass.set(project.application.mainClass.isPresent() ? project.application.mainClass.get() : project.application.mainClassName)
+                        mainClass.set(project.springBoot.mainClass)
```

Comment:
Ooh, this is added as in https://github.com/spring-projects/spring-boot/blob/fe9864847f3e3c4077a207ee1b31a10a078ec4a8/build-plugin/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/plugin/SpringBootPlugin.java#L122 ?

Reply:
the diff is deceiving here. this code has not changed